draft-thomson-http-oblivious.md

```diff
@@ -264,15 +306,131 @@ Message encrypted message.
 Plaintext Messages support arbitrary length padding. Clients and servers MAY pad HTTP messages
 as needed to hide metadata leakage through ciphertext length.
 
-# Responsibility of Roles
+
+# Responsibility of Roles {#trust}
+
+In this design, a client wishes to make a request of a server that is
+authoritative for the oblivious target resource. The client wishes to make this
+request without linking that request with either:
+
+1. The identity at the network and transport layer of the client (that is, the
+   client IP address and TCP or UDP port number the client uses to create a
+   connection).
+
+2. Any other request the client might have made in the past or might make in
+   the future.
+
+In order to ensure this, the client selects a proxy (that serves the oblivious
+proxy resource) that it trusts will protect this information by forwarding the
+encapsulated request and response without passing the server (that serves the
+oblivious request resource).
+
+In this section, a deployment where there are three entities is considered:
+
+* A client makes requests and receives responses
+* A proxy operates the oblivious proxy resource
+* A server operates both the oblivious request resource and the oblivious
+  target resource
+
+To achieve the stated privacy goals, the oblivious proxy resource cannot be
+operated by the same entity as the oblivious request resource. However,
+colocation of the oblivious request resource and oblivious target resource
+simplifies the interactions between those resources without affecting client
+privacy.
+
 
 ## Client
 
-## Oblivious Proxy Resource
+Clients have the fewest direct responsibilities, though clients do need to
+ensure that they do not undermine the process.
+
+Clients cannot carry connection-level state between requests as they only
+establish direct connections to the proxy responsible for the oblivious proxy
+resource. However, clients need to ensure that they construct requests without
+any information gained from previous requests. Otherwise, the server might be
+able to use that information to link requests. Cookies {{?COOKIES=RFC6265}} are
+the most obvious feature that MUST NOT be used by clients. However, clients
+need to include all information learned from requests, which could include the
+identity of resources.
+
+Clients also need to ensure that they correctly generate a new HPKE context for
+every request, using a good source of entropy ({{?RANDOM=RFC4086}}). Key reuse
+not only risks linkability, but it could expose request and response contents
+to the proxy. Note that an HPKE implementation is stateful and so prevents AEAD
+nonce reuse.
```

Comment:
Since contexts aren't re-used, I don't think we need to say this.

```suggestion
to the proxy. 
```

Reply:
I was thinking that this was something we could say because it explains that reuse of a context doesn't necessarily lead to this problem.